File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/util/SharedStringLocalization.kt
package com.mbta.tid.mbta_app.android.util

import androidx.compose.runtime.Composable
import androidx.compose.ui.res.stringResource
import com.mbta.tid.mbta_app.android.R
import com.mbta.tid.mbta_app.utils.SharedString

val SharedString.value: String
    @Composable
    get() =
        when (this) {
            SharedString.CommuterRailAndFerryTickets ->
                stringResource(R.string.resources_link_mticket)
            SharedString.DebugMode -> stringResource(R.string.feature_flag_debug_mode)
            SharedString.FareInformation -> stringResource(R.string.resources_link_fare_info)
            SharedString.FeatureFlagsSection -> stringResource(R.string.more_section_feature_flags)
            SharedString.MapDisplay -> stringResource(R.string.setting_toggle_map_display)
            SharedString.MTicketApp -> stringResource(R.string.resources_link_mticket_note)
            SharedString.Notifications -> "Notifications" // Temp feature flag
            SharedString.PrivacyPolicy -> stringResource(R.string.other_link_privacy_policy)
            SharedString.ResourcesSection -> stringResource(R.string.more_section_resources)
            SharedString.RouteSearch -> stringResource(R.string.feature_flag_route_search)
            SharedString.SendAppFeedback -> stringResource(R.string.feedback_link_form)
            SharedString.SettingsSection -> stringResource(R.string.more_section_settings)
            SharedString.SoftwareLicenses -> stringResource(R.string.software_licenses)
            SharedString.StationAccessibilityInfo ->
                stringResource(R.string.setting_station_accessibility)
            SharedString.SupportAccessibilityNote ->
                stringResource(R.string.more_section_support_note_accessibility)
            SharedString.SupportHours -> stringResource(R.string.more_section_support_note_hours)
            SharedString.SupportSection -> stringResource(R.string.more_section_support)
            SharedString.TermsOfUse -> stringResource(R.string.other_link_tos)
            SharedString.TripPlanner -> stringResource(R.string.resources_link_trip_planner)
            SharedString.ViewSourceOnGithub -> stringResource(R.string.other_link_source_code)
        }

File: iosApp/iosApp/Utils/Extensions/SharedStringLocalization.swift
//
//  SharedStringLocalization.swift
//  iosApp
//
//  Created by Brandon Rodriguez on 10/1/25.
//  Copyright © 2025 MBTA. All rights reserved.
//

import Foundation
import Shared

extension SharedString {
    var value: String {
        switch self {
        case .commuterRailAndFerryTickets:
            NSLocalizedString(
                "Commuter Rail and Ferry Tickets",
                comment: "Label for a More page link to the MBTA mTicket app"
            )
        case .debugMode:
            NSLocalizedString(
                "Debug Mode",
                comment: "A setting on the More page to display debug information (only visible for developers)"
            )
        case .fareInformation:
            NSLocalizedString(
                "Fare Information",
                comment: "Label for a More page link to fare information on MBTA.com"
            )
        case .featureFlagsSection:
            NSLocalizedString(
                "Feature Flags",
                comment: "More page section header, only displayed in the developer app for enabling in-progress features"
            )
        case .mapDisplay:
            NSLocalizedString(
                "Map Display",
                comment: "A setting on the More page to show / hide maps from the app"
            )
        case .mticketApp:
            NSLocalizedString(
                "mTicket App",
                comment: "Footnote underneath the \"Commuter Rail and Ferry Tickets\" label on the More page link to the MBTA mTicket app"
            )
        case .notifications:
            "Notifications" // Temp feature flag
        case .privacyPolicy:
            NSLocalizedString(
                "Privacy Policy",
                comment: "Label for a More page link to the MBTA.com privacy policy"
            )
        case .resourcesSection:
            NSLocalizedString(
                "Resources",
                comment: "More page section header, includes links to MBTA.com and mTicket app"
            )
        case .routeSearch:
            NSLocalizedString(
                "Route Search",
                comment: "A setting on the More page to display routes in search (only visible for developers)"
            )
        case .sendAppFeedback:
            NSLocalizedString(
                "Send App Feedback",
                comment: "Label for a More page link to a form to provide feedback on the app itself"
            )
        case .settingsSection:
            NSLocalizedString(
                "Settings",
                comment: "More page section header, includes settings that the user can configure"
            )
        case .softwareLicenses:
            NSLocalizedString(
                "Software Licenses",
                comment: "Label for a More page link to view dependency licenses"
            )
        case .stationAccessibilityInfo:
            NSLocalizedString(
                "Station Accessibility Info",
                comment: "A setting on the More page to toggle displaying station accessibility info"
            )
        case .supportAccessibilityNote:
            NSLocalizedString(
                "711 for TTY callers; VRS for ASL callers",
                comment: "Footnote under the More page support phone number, these are the instructions for accessible support via teletypewriter or video relay service"
            )
        case .supportHours:
            NSLocalizedString(
                "Monday through Friday: 6:30 AM - 8 PM",
                comment: "Footnote under the More page support header, these are the hours for the support call center"
            )
        case .supportSection:
            NSLocalizedString(
                "General MBTA Information & Support",
                comment: "More page section header, includes user support resources"
            )
        case .termsOfUse:
            NSLocalizedString(
                "Terms of Use",
                comment: "Label for a More page link to the MBTA.com terms of use"
            )
        case .tripPlanner:
            NSLocalizedString(
                "Trip Planner",
                comment: "Label for a More page link to the MBTA.com trip planner"
            )
        case .viewSourceOnGithub:
            NSLocalizedString(
                "View Source on GitHub",
                comment: "Label for a More page link to the MBTA Go source code"
            )
        }
    }
}
